### ISSUE TITLE
On dragable sort receive error - can't get model of null

### ISSUE THREAD

**Issue by chiz-developer** [optimistdigital/nova-sortable#92]
Solved by providing viaResourceId to findResourceOrFail
Can you fix this?
![image](https://user-images.githubusercontent.com/13472539/117355523-5d28bb80-aebb-11eb-8145-c46b157fec18.png)

**Comment by chiz-developer** [comment]
$validationResult is null here
![image](https://user-images.githubusercontent.com/13472539/117355797-ac6eec00-aebb-11eb-8189-1c3fd4a5fab2.png)

**Comment by Tarpsvo** [comment]
I need to know more about your setup. What relationship are you using, what config etc?

**Comment by chiz-developer** [comment]
Laravel Framework 7.29.3
Nova 3

HasMany relation

**Comment by Tarpsvo** [comment]
> Solved by providing viaResourceId to findResourceOrFail
> Can you fix this?
> ![image](https://user-images.githubusercontent.com/13472539/117355523-5d28bb80-aebb-11eb-8145-c46b157fec18.png)

That's weird, it shouldn't fix it, as resourceId is null for that reorder request.

Can you see what exception you have on line 25? Ie:
```
        try {
            if ($resource === null) {
                $resource = isset($request->resourceId) ? $request->findResourceOrFail() : $request->newResource();
            }
        } catch (\Exception $e) {
            throw $e;
            // return null;
        }
```

I honestly can't seem to reproduce this error.

**Comment by chiz-developer** [comment]
Trying to get property 'model' of non-object {"userId":1,"exception":"[object] (ErrorException(code: 0): Trying to get property 'model' of non-object at /var/www/laravelproject/api/vendor/optimistdigital/nova-sortable/src/Http/Controllers/SortableController.php:14)
[stacktrace]
#0 /var/www/laravelproject/api/vendor/optimistdigital/nova-sortable/src/Http/Controllers/SortableController.php(14): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(48): OptimistDigital\\NovaSortable\\Http\\Controllers\\SortableController->updateOrder()
#2 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Route.php(239): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#3 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Route.php(196): Illuminate\\Routing\\Route->runController()
#4 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(685): Illuminate\\Routing\\Route->run()
#5 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#6 /var/www/laravelproject/api/vendor/laravel/nova/src/Http/Middleware/Authorize.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#7 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\Authorize->handle()
#8 /var/www/laravelproject/api/vendor/laravel/nova/src/Http/Middleware/BootTools.php(20): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#9 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\BootTools->handle()
#10 /var/www/laravelproject/api/vendor/laravel/nova/src/Http/Middleware/DispatchServingNovaEvent.php(20): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#11 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\DispatchServingNovaEvent->handle()
#12 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#13 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#14 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php(44): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#15 /var/www/laravelproject/api/vendor/laravel/nova/src/Http/Middleware/Authenticate.php(31): Illuminate\\Auth\\Middleware\\Authenticate->handle()
#16 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\Authenticate->handle()
#17 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(77): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#18 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#19 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#20 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#21 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#22 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(62): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#23 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#24 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#25 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#26 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#27 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#28 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#29 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(687): Illuminate\\Pipeline\\Pipeline->then()
#30 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(662): Illuminate\\Routing\\Router->runRouteWithinStack()
#31 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(628): Illuminate\\Routing\\Router->runRoute()
#32 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(617): Illuminate\\Routing\\Router->dispatchToRoute()
#33 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(165): Illuminate\\Routing\\Router->dispatch()
#34 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#35 /var/www/laravelproject/api/vendor/laravel/nova/src/Http/Middleware/ServeNova.php(26): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#36 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\ServeNova->handle()
#37 /var/www/laravelproject/api/app/Http/Middleware/Sanitizer.php(31): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#38 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): App\\Http\\Middleware\\Sanitizer->handle()
#39 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#41 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#42 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#43 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#44 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#45 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(63): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#46 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode->handle()
#47 /var/www/laravelproject/api/vendor/fruitcake/laravel-cors/src/HandleCors.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#48 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle()
#49 /var/www/laravelproject/api/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#50 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#51 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#52 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(140): Illuminate\\Pipeline\\Pipeline->then()
#53 /var/www/laravelproject/api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(109): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#54 /var/www/laravelproject/api/public/index.php(55): Illuminate\\Foundation\\Http\\Kernel->handle()
#55 /var/www/laravelproject/api/server.php(21): require_once('/var/www/builda...')
#56 {main}
"}

**Comment by Tarpsvo** [comment]
That's the error thrown when $validationResult is null, what about the error at HasSortableRows.php at line 25? Thanks!

**Comment by chiz-developer** [comment]
![image](https://user-images.githubusercontent.com/13472539/117451350-1a162900-af4b-11eb-8de0-c07745e2f07a.png)
![image](https://user-images.githubusercontent.com/13472539/117451401-28fcdb80-af4b-11eb-9d16-29d6600a2ad8.png)
![image](https://user-images.githubusercontent.com/13472539/117451426-3023e980-af4b-11eb-8bb8-61d075736ab1.png)

**Comment by Tarpsvo** [comment]
Ah interesting, so on the server, it somehow becomes an empty string. I have no idea how that happens though.

I replaced the `isset()` with `!empty()` which also checks for `0` and `''` and released version 2.3.1 with the fix. Thanks a ton for your help! Good luck!

**Comment by chiz-developer** [comment]
thank you too!
